src/components/pages/VisualisationEditor.tsx

```diff
@@ -33,15 +37,19 @@ const VisualisationEditor: React.FC = () => {
         return <Text>Loading…</Text>;
     }
 
-    if (!data || !data.allMetadata.some((el) => el.id === id)) {
+    if (!data || (!data.allMetadata.some((el) => el.id === id) && !visualisation)) {
         // No metadata was found. Go back
         history.push('/explore');
         return null;
     }
 
-    // Safe non-null-assertion due to the .some above
-    // eslint-disable-next-line
-    const metadata = data.allMetadata.find((el) => el.id === id)!;
+    const metadata = visualisation
+        ? data.allMetadata.find((el) => el.id === visualisation.metadataId)
```

Comment:
Could you compare this with `el.dataSourceId` and similar on visualisation instea

Reply:
Yes sir